### ISSUE TITLE
Debugging and update examples.

### ISSUE THREAD

**Issue by cedlemo** [garrigue/lablgtk#4]
Here is a list of the examples to test the lablgtk2on3 (#2 ) ordered on their number of lines. It will help to keep track of what has been done and what remains to do. 

-  [ ]   (19) examples/curve.ml
-  [ ]   (19) examples/entry2.ml
-  [ ]   (19) examples/seppala.ml
-  [ ]   (25) examples/socket.ml
-  [x]   (27) examples/hello.ml
-  [ ]   (28) examples/spin.ml
-  [ ]   (30) examples/calendar.ml
-  [ ]   (30) examples/drawing.ml
-  [ ]   (30) examples/runthread.ml
-  [ ]   (32) examples/combo.ml
-  [ ]   (32) examples/link_button.ml
-  [ ]   (33) examples/expander.ml
-  [ ]   (33) examples/nihongo.ml
-  [ ]   (36) examples/about.ml
-  [ ]   (36) examples/assistant.ml
-  [ ]   (37) examples/dialog-thread.ml
-  [ ]   (39) examples/events2.ml
-  [ ]   (39) examples/fixpoint.ml
-  [ ]   (39) examples/giotest.ml
-  [ ]   (39) examples/notebook.ml
-  [ ]   (41) examples/pixview.ml
-  [ ]   (42) examples/buttons.ml
-  [ ]   (43) examples/text/unicode_viewer.ml
-  [ ]   (44) examples/GL/simple.ml
-  [ ]   (44) examples/progressbar.ml
-  [ ]   (44) examples/scrolledwin.ml
-  [ ]   (45) examples/counter.ml
-  [ ]   (45) examples/glade/glade_demo.ml
-  [ ]   (45) examples/rsvg/test_rsvg.ml
-  [ ]   (46) examples/signal_override.ml
-  [ ]   (47) examples/GL/simple_th.ml
-  [ ]   (48) examples/cputs.ml
-  [ ]   (48) examples/radiobuttons.ml
-  [ ]   (49) examples/text/unicode_table.ml
-  [ ]   (57) examples/label.ml
-  [ ]   (57) examples/lissajous.ml
-  [ ]   (59) examples/entry.ml
-  [ ]   (59) examples/sourceview/sourceview2.ml
-  [ ]   (61) examples/clist.ml
-  [ ]   (63) examples/cgets.ml
-  [ ]   (69) examples/gioredirect.ml
-  [ ]   (70) examples/slide_show.ml
-  [ ]   (71) examples/filechooser.ml
-  [ ]   (78) examples/GL/aargb.ml
-  [ ]   (78) examples/eventbox.ml
-  [ ]   (78) examples/glade/gladecalc.ml
-  [ ]   (79) examples/canvas/canvas-richtext.ml
-  [ ]   (81) examples/canvas/canvas-features.ml
-  [ ]   (83) examples/canvas/toroid.png
-  [ ]   (85) examples/tree.ml
-  [ ]   (86) examples/spell.ml
-  [ ]   (92) examples/test.xpm
-  [ ]   (94) examples/entrycompletion.ml
-  [ ]  (104) examples/events.ml
-  [ ]  (106) examples/GL/tutorial-2.ml
-  [ ]  (106) examples/sourceview/test2.ml
-  [ ]  (107) examples/fifteen.ml
-  [ ]  (107) examples/image.ml
-  [ ]  (112) examples/editor.ml
-  [ ]  (113) examples/druid.ml
-  [ ]  (113) examples/sourceview/test.ml
-  [ ]  (114) examples/GL/tutorial-3.ml
-  [ ]  (123) examples/calc.ml
-  [ ]  (123) examples/sourceview/example2.ml
-  [ ]  (124) examples/combobox.ml
-  [ ]  (125) examples/GL/tutorial-4.ml
-  [ ]  (131) examples/GL/texturesurf.ml
-  [ ]  (132) examples/GL/scene.ml
-  [ ]  (132) examples/sourceview/ocaml.lang
-  [ ]  (133) examples/panel/applet.ml
-  [ ]  (134) examples/editor2.ml
-  [ ]  (134) examples/rpn.ml
-  [ ]  (136) examples/GL/planet.ml
-  [ ]  (141) examples/kaimono.ml
-  [ ]  (144) examples/tooltip.ml
-  [ ]  (145) examples/canvas/canvas-fifteen.ml
-  [ ]  (146) examples/assistant_tutorial.ml
-  [ ]  (146) examples/timer.ml
-  [ ]  (153) examples/action.ml
-  [ ]  (153) examples/iconview.ml
-  [ ]  (173) examples/tree_model.ml
-  [ ]  (176) examples/canvas/canvas-curve.ml
-  [ ]  (179) examples/GL/tutorial-5.ml
-  [ ]  (184) examples/text/test_text.ml
-  [ ]  (185) examples/custom_list_generic.ml
-  [ ]  (192) examples/custom_tree.ml
-  [ ]  (199) examples/tree_store.ml
-  [ ]  (207) examples/text/test.txt
-  [ ]  (210) examples/accel_tree.ml
-  [ ]  (210) examples/pousse.ml
-  [ ]  (212) examples/tron.ml
-  [ ]  (241) examples/csview.ml
-  [ ]  (260) examples/dcalendar.ml
-  [ ]  (264) examples/custom_tree_generic.ml
-  [ ]  (266) examples/GL/gears.ml
-  [ ]  (277) examples/toolbar.ml
-  [ ]  (283) examples/text/text-demo.ml
-  [ ]  (291) examples/fixed_editor.ml
-  [ ]  (331) examples/canvas/canvas-arrowhead.ml
-  [ ]  (346) examples/glade/project2.glade
-  [ ]  (453) examples/canvas/canvas-primitives.ml
-  [ ]  (518) examples/testdnd.ml
-  [ ]  (618) examples/GL/morph3d.ml
-  [ ] (1130) examples/testgtk.ml

**Comment by herbelin** [comment]
Hi @cedlemo, I'm currently working on the lablgtk2on3 branch in order to port Coq's UI CoqIDE to gtk3. Note that some examples are failing only because `GMain.init` is now required prior to the first class creation. I did not make an exhaustive check, but this is the case for instance of `iconview.ml` and `tree_model.ml`, which are respectively fixed with the following patches:
```
diff --git a/examples/iconview.ml b/examples/iconview.ml
index 21e7b74..7daf40d 100644
--- a/examples/iconview.ml
+++ b/examples/iconview.ml
@@ -146,6 +146,7 @@ let do_iconview window =
   | _ -> ()
   
 let main = 
+  GMain.init ();
   let w = GWindow.window ~title:"GtkIconView demo" ~width:650 ~height:400 () in
   w#connect#destroy GMain.quit ;
   do_iconview w ;
diff --git a/examples/tree_model.ml b/examples/tree_model.ml
index 9741bdc..8d1d490 100644
--- a/examples/tree_model.ml
+++ b/examples/tree_model.ml
@@ -151,6 +151,7 @@ let inspect_data_2 column (model : GTree.model) =
 
        
 let main =
+  GMain.init ();
   let w = GWindow.window ~title:"GtkListStore test" () in
   w#connect#destroy GMain.quit ;
 
```

**Comment by garrigue** [comment]
@herbelin I don't think this is new. Examples are intended to be run using the lablgtk top-level, which links `gtkInit.cmo`, whose role is just to call the `GMain.init` function.
How were you trying to run the tests?

The reason the call to the init function is delayed was originally to allow Unison to link with lablgtk but still be allowed to run in a shell window in text mode.

**Comment by herbelin** [comment]
> How were you trying to run the tests?

I did not know the intended way to run them. I compiled them directly with `ocamlc`.

The same invocation indeed already fails with lablgtk2, sorry. (I still have to understand why I had to move the call to `GMain.init` earlier in CoqIDE though.)

**Comment by XVilka** [comment]
@cedlemo it is probably worth to update the list, I think.

**Comment by cedlemo** [comment]
@XVilka, yep.

**Comment by XVilka** [comment]
@cedlemo was there any progress? There were a lot of changes since the list was formed initially. And lablgtk3 is now outlived a few beta releases already.

**Comment by XVilka** [comment]
See also this PR https://github.com/garrigue/lablgtk/pull/84
